test/cassandrax/keyspace_test.exs

```diff
@@ -0,0 +1,377 @@
+      changeset3 = TestKeyspace.get(TestData, id: "1") |> Changeset.change(value: "one last one")
+      expectation = ["new one", "last one", "one last one"]
+      TestKeyspace.batch(fn batch ->
+        batch
+        |> TestKeyspace.batch_update(changeset1)
+        |> TestKeyspace.batch_update(changeset2)
+        |> TestKeyspace.batch_update(changeset3)
+      end)
+      assert TestKeyspace.one(where(TestData, id: "1")).value in expectation
+    end
+
+    test "update set data of the same records" do
+      changeset1 = TestKeyspace.get(TestData, id: "1") |> Changeset.change(svalue: MapSet.new(["hello world"]))
+      changeset2 = TestKeyspace.get(TestData, id: "1") |> Changeset.change(svalue: MapSet.new(["pandemic world"]))
+      expectation2 = %{@one | svalue: MapSet.new(["hello world", "pandemic world"])}
+      TestKeyspace.batch(fn batch ->
+        batch
+        |> TestKeyspace.batch_update(changeset1)
+        |> TestKeyspace.batch_update(changeset2)
+      end)
+      assert TestKeyspace.one(where(TestData, id: "1")) == expectation2
+    end
+
+    test "single delete" do
+      TestKeyspace.batch(fn batch ->
+        data = TestKeyspace.get(TestData, id: "0")
+        batch
+        |> TestKeyspace.batch_delete(data)
+      end)
+
+      assert TestKeyspace.one(where(TestData, id: "0")) == nil
+    end
+
+    test "multiple deletes" do
+      TestKeyspace.batch(fn batch ->
+        batch
+        |> TestKeyspace.batch_delete(@zero)
+        |> TestKeyspace.batch_delete(@one)
+      end)
+
+      refute list_set_includes?(TestKeyspace.all(TestData), [@zero,  @one])
+    end
+
+    test "insert and update" do
+      changeset = TestKeyspace.get(TestData, id: "1") |> Changeset.change(value: "new one")
+      expectation = %{@one | value: "new one"}
+      TestKeyspace.batch(fn batch ->
+        batch
+        |> TestKeyspace.batch_insert(@four)
+        |> TestKeyspace.batch_update(changeset)
+      end)
+
+      assert list_set_includes?(TestKeyspace.all(TestData), [@four, expectation])
+    end
+
+    test "delete then update" do
+      changeset = TestKeyspace.get(TestData, id: "1") |> Changeset.change(value: "new one")
+      expectation = [@one, %{@one | value: "new one"}, nil]
+      TestKeyspace.batch(fn batch ->
+        batch
+        |> TestKeyspace.batch_delete(@one)
+        |> TestKeyspace.batch_update(changeset)
+      end)
+
+      assert TestKeyspace.one(where(TestData, id: "1")) in expectation
+    end
+   end
+
+  describe "cql" do
+    @describetag seeds: [@zero, @one, @two]
+
+    test "valid cql" do
+      statement = """
+      SELECT * FROM #{TestKeyspace.__keyspace__()}.test_data
+      """
+      assert {:ok, _} = Cassandrax.cql(TestConn, statement)
+
+      statement = [
+        "SELECT value FROM ",
+        "#{TestKeyspace.__keyspace__()}.test_data ",
+        "WHERE id = '1'"
+      ]
+      assert {:ok, _} = Cassandrax.cql(TestConn, statement)
+    end
```

Comment:
why did you choose not to assert the results?

Reply:
I was checking the validity of the cql, I'll add the tests for the results too! :)

Reply:
thanks! this was more a question than a change request, but thanks for adding those